packages/instant/src/components/buy_button.tsx

```diff
@@ -73,7 +73,9 @@ export class BuyButton extends React.Component<BuyButtonProps> {
             throw e;
         }
 
-        this.props.onBuyProcessing(buyQuote, txHash);
+        const startTimeUnix = new Date().getTime();
+        const expectedEndTimeUnix = startTimeUnix + gasInfo.estimatedTimeMs;
```

Comment:
If it weren't for us only having access to `estimatedTimeMs` here, I would say this kind of logic belongs in the reducer. How do you feel about just passing `estimatedTime` up through the callbacks, and having the `Date.now()` call in the reducer? I guess `startTimeUnix` is more accurate if done here..

Reply:
I'd prefer to leave as-is as it's a simple & quick calculation, it's nice to grab the startTime immediately, and this is the only place where `onBuyProcessing` is triggered, but can change if you feel strongly

Reply:
Yea agreed